source/parser.hera

```diff
@@ -7523,6 +7523,19 @@ Reset
         // [indent, statement]
         module.prelude.push(["", [preludeVar, isRef, typeSuffix, " = Object.is", asAny, ";\n"]])
       },
+      /**
+       * Array length check with type guard.
+       * From bbrk24 https://github.com/DanielXMoore/Civet/issues/1075#issuecomment-1971897483
```

Comment:
As much as I appreciate credit, notice the Discord link in that comment -- I didn't come up with this.